paimon-core/src/main/java/org/apache/paimon/schema/Schema.java

```diff
@@ -133,6 +155,12 @@ private static List<DataField> normalizeFields(
         return newFields;
     }
 
+    private static Set<String> duplicate(List<String> names) {
```

Comment:
Could we return the result that uses `anyMatch(name -> Collections.frequency(names, name) > 1)` that represents whether the `names` is duplicate? Meanwhile uses the return value to invoke `Preconditions.checkState`.

Reply:
I collect duplicate columns to put them in error message:
`Found: {duplicatedColumns}`. 
Maybe it is not necessary. What's your suggestion?